support-frontend/assets/helpers/existingPaymentMethods/existingPaymentMethods.js

```diff
@@ -17,24 +17,23 @@ export type ExistingPaymentMethodSubscription = {
 
 type ExistingPaymentType = 'Card' | 'DirectDebit';
 
-export type PartialExistingPaymentMethod = {
-  paymentType: ExistingPaymentType;
+export type NotRecentlySignedInExistingPaymentMethod = {
+  paymentType: ExistingPaymentType,
 };
 
-export type FullDetailExistingPaymentMethod = {
+export type RecentlySignedInExistingPaymentMethod = {
+  paymentType: ExistingPaymentType,
   billingAccountId: string,
   subscriptions: ExistingPaymentMethodSubscription[],
   card?: string,
   mandate?: string,
-  // can't use Intersection types to avoid repeating this as it breaks isFullDetailExistingPaymentMethod because; Flow!
-  paymentType: ExistingPaymentType
 };
 
-export type ExistingPaymentMethod = PartialExistingPaymentMethod | FullDetailExistingPaymentMethod;
+export type ExistingPaymentMethod = NotRecentlySignedInExistingPaymentMethod | RecentlySignedInExistingPaymentMethod;
 
 // ----- Functions ----- //
 
-function isFullDetailExistingPaymentMethod(existingPaymentMethod: ExistingPaymentMethod): boolean %checks {
+function isUsableExistingPaymentMethod(existingPaymentMethod: ExistingPaymentMethod): boolean %checks {
```

Comment:
I hadn't seen %checks or the double bang to return a boolean before, interesting!